### PR TITLE
The v2 migration script fix - migrate Fluent Bit image key instead of deleting it

### PR DESCRIPTION
**Pull request by perk-sumo**
Convert this: https://github.com/helm/charts/blob/master/stable/fluent-bit/values.yaml#L5
```
image:
  fluent_bit:
    repository: fluent/fluent-bit
    tag: 1.3.7
  pullPolicy: Always
```

into this: https://github.com/fluent/helm-charts/blob/fb1f611236a3d4b4f56514bb1d05b23820f8a409/charts/fluent-bit/values.yaml#L9
```
image:
  repository: fluent/fluent-bit
  tag: 1.3.7
  pullPolicy: Always
```

### CODE REVIEW

**Review by pmalek-sumo — APPROVED** [review]

File: tests/upgrade_v2_script/static/fluent_bit.output.yaml [hunk 1]
```diff
@@ -36,6 +36,9 @@ fluent-bit:
     - effect: NoSchedule
       operator: Exists
   image:
+    fluent_bit:
```
Comment (astencel-sumo) [inline tests/upgrade_v2_script/static/fluent_bit.output.yaml:39]: Shouldn't this be migrated?

File: deploy/helm/sumologic/upgrade-2.0.0.sh [hunk 2]
```diff
@@ -15,6 +15,7 @@ readonly MAX_YQ_VERSION=4.0.0
 readonly KEY_MAPPINGS="
 prometheus-operator.prometheusOperator.tlsProxy.enabled:kube-prometheus-stack.prometheusOperator.tls.enabled
 otelcol.deployment.image.name:otelcol.deployment.image.repository
+fluent-bit.image.fluent_bit:fluent-bit.image.repository
```
Comment (astencel-sumo) [inline deploy/helm/sumologic/upgrade-2.0.0.sh:18]: Is this correct? I thought `fluent-bit.image.fluent_bit` maps to `fluent-bit.image` and not to `fluent-bit.image.repository`?

File: tests/upgrade_v2_script/static/fluent_bit.output.yaml (hunk 1 above)
Comment (astencel-sumo) [reply tests/upgrade_v2_script/static/fluent_bit.output.yaml:39]: But the tests on the PR passed, this means migration script tests are not run as part of PR check?

Comment (pmalek-sumo) [reply tests/upgrade_v2_script/static/fluent_bit.output.yaml:39]: They are https://github.com/SumoLogic/sumologic-kubernetes-collection/blob/eb4d7a52585ec7e6fadfae55160509731075b61f/ci/tests.sh#L19-L21

Something's not right

File: deploy/helm/sumologic/upgrade-2.0.0.sh (hunk 2 above)
Comment (perk-sumo) [reply deploy/helm/sumologic/upgrade-2.0.0.sh:18]: Good call, it's wrong. Let me fix that.

File: tests/upgrade_v2_script/static/fluent_bit.output.yaml (hunk 1 above)
Comment (perk-sumo) [reply tests/upgrade_v2_script/static/fluent_bit.output.yaml:39]: Yeah, it's
https://github.com/helm/charts/blob/master/stable/fluent-bit/values.yaml#L5
vs
https://github.com/fluent/helm-charts/blob/fb1f611236a3d4b4f56514bb1d05b23820f8a409/charts/fluent-bit/values.yaml#L9

Comment (perk-sumo) [reply tests/upgrade_v2_script/static/fluent_bit.output.yaml:39]: I mean - for the output the `fluent_bit` is still there, so this one should be correct (let me double check).
It's just not there for the Helm chart value.

Comment (perk-sumo) [reply tests/upgrade_v2_script/static/fluent_bit.output.yaml:39]: Hm, nope, I'm wrong 😅

Comment (perk-sumo) [reply tests/upgrade_v2_script/static/fluent_bit.output.yaml]: Tests fixed.

File: deploy/helm/sumologic/upgrade-2.0.0.sh (hunk 2 above)
Comment (sumo-drosiek) [reply deploy/helm/sumologic/upgrade-2.0.0.sh:18]: Please update tests according to change :sweat_smile:

File: tests/upgrade_v2_script/static/fluent_bit.output.yaml [hunk 3]
```diff
@@ -36,7 +36,9 @@ fluent-bit:
     - effect: NoSchedule
       operator: Exists
   image:
-    pullPolicy: IfNotPresent
+    repository: public.ecr.aws/sumologic/fluent-bit
+    tag: 1.6.0
```
Comment (sumo-drosiek) [inline tests/upgrade_v2_script/static/fluent_bit.output.yaml:40]: should be `1.6.10`

File: tests/upgrade_v2_script/static/fluent_bit_no_image.input.yaml [hunk 4]
```diff
@@ -0,0 +1,155 @@
...
+      Systemd_Filter  _SYSTEMD_UNIT=set-aws-environment.service
+      Systemd_Filter  _SYSTEMD_UNIT=system-cloudinit.service
+      Systemd_Filter  _SYSTEMD_UNIT=systemd-timesyncd.service
+      Systemd_Filter  _SYSTEMD_UNIT=update-ca-certificates.service
+      Systemd_Filter  _SYSTEMD_UNIT=user-cloudinit.service
+      Systemd_Filter  _SYSTEMD_UNIT=var-lib-etcd2.service
+      Max_Entries     1000
+      Read_From_Tail  true
+
+    @INCLUDE fluent-bit-output.conf
```
Comment (sumo-drosiek) [inline tests/upgrade_v2_script/static/fluent_bit_no_image.input.yaml:155]: Imho we should check only image migration, not full fluent-bit config

File: tests/upgrade_v2_script/static/fluent_bit.output.yaml (hunk 3 above)
Comment (perk-sumo) [reply tests/upgrade_v2_script/static/fluent_bit.output.yaml:40]: This should be whatever is in the input file (which is `1.6.0`)

File: tests/upgrade_v2_script/static/fluent_bit_no_image.input.yaml (hunk 4 above)
Comment (perk-sumo) [reply tests/upgrade_v2_script/static/fluent_bit_no_image.input.yaml]: Good point, I deleted most of the sections (left some of them).

File: tests/upgrade_v2_script/static/fluent_bit.output.yaml (hunk 3 above)
Comment (sumo-drosiek) [reply tests/upgrade_v2_script/static/fluent_bit.output.yaml:40]: what if it is same value like in our old `values.yaml`? IMHO we should migrate it

Comment (sumo-drosiek) [reply tests/upgrade_v2_script/static/fluent_bit.output.yaml:40]: and 1.6.0 is corrupted

File: tests/upgrade_v2_script/static/fluent_bit_no_image.input.yaml (hunk 4 above)
Comment (perk-sumo) [reply tests/upgrade_v2_script/static/fluent_bit_no_image.input.yaml]: Also I added another testcase just for the image section.

File: tests/upgrade_v2_script/static/fluent_bit.output.yaml (hunk 3 above)
Comment (perk-sumo) [reply tests/upgrade_v2_script/static/fluent_bit.output.yaml:40]: I don't agree, we should not change the values. Let me add info about new default value though.

Comment (sumo-drosiek) [reply tests/upgrade_v2_script/static/fluent_bit.output.yaml:40]: waiting.. :D

File: deploy/helm/sumologic/upgrade-2.0.0.sh (hunk 2 above)
Comment (perk-sumo) [reply deploy/helm/sumologic/upgrade-2.0.0.sh]: Done.

File: tests/upgrade_v2_script/static/fluent_bit.output.yaml (hunk 3 above)
Comment (perk-sumo) [reply tests/upgrade_v2_script/static/fluent_bit.output.yaml:40]: Done :)

File: tests/upgrade_v2_script/static/fluent_bit_no_image.input.yaml (hunk 4 above)
Comment (perk-sumo) [reply tests/upgrade_v2_script/static/fluent_bit_no_image.input.yaml]: Done.

**Review by sumo-drosiek — APPROVED** [review]

**Pull request merged**